yt/yql/agent/type_builder.cpp

```diff
@@ -0,0 +1,334 @@
+#include "type_builder.h"
+
+namespace NYT::NYqlAgent {
+
+using namespace NTableClient;
+
+////////////////////////////////////////////////////////////////////////////////
+
+TTypeBuilder::TTypeBuilder()
+{
+    ItemsStack.emplace(TItemType());
+}
+
+TLogicalTypePtr TTypeBuilder::PullResult() {
+    return Pop();
+}
+
+
+template<class T>
```

Comment:
Why do we need template here?

Reply:
Because it is the stack of variants. 

https://github.com/ytsaurus/ytsaurus/pull/967/files#diff-13ea3bafc5e31ddbf479f85866a7729c7a012e525ec44ac37f71a01e04627a1cR255